SPO/README.md

```diff
@@ -0,0 +1,103 @@
+# How to create a SPFx extension solution with AppInsights installed from scratch
+
+## Set up Application Insights 
+1. Follow [SPFx Extensions instruction](https://docs.microsoft.com/en-us/sharepoint/dev/spfx/extensions/get-started/build-a-hello-world-extension) to create a SharePoint Framework Extension.
+2. Install Application Insights via [NPM setup](https://github.com/microsoft/ApplicationInsights-JS#getting-started). 
+3. Under your Extension solution src folder, update ApplicationCustomizer.ts file:
+    a. [Set up Application Insights](https://github.com/microsoft/ApplicationInsights-JS#npm-setup-ignore-if-using-snippet-setup)
+    ```js
+    import { ApplicationInsights } from '@microsoft/applicationinsights-web'
+
+    const appInsights = new ApplicationInsights({ config: {
+    instrumentationKey: 'YOUR_INSTRUMENTATION_KEY_GOES_HERE'
+    /* ...Other Configuration Options... */
+    } });
+    appInsights.loadAppInsights();
+    appInsights.trackPageView(); // Manually call trackPageView to establish the current user/session/pageview
+    ```
+
+    b. Expose all configuration options you need in the ApplicationCustomizerProperties interface.
+    ```js
+    export interface IAppInsightsApplicationCustomizerProperties {
+        /* ...Other Configuration Options... */
+        instrumentationKey: string;
+    }
+    ```
+    c. Update appInsights instance config to map with exposed properties.
+    ```js
+    import { ApplicationInsights } from '@microsoft/applicationinsights-web'
+
+    const appInsights = new ApplicationInsights({ config: {
+    instrumentationKey: this.properties.instrumentationKey
+    /* ...Other Configuration Options... */
+    } });
+    appInsights.loadAppInsights();
+    appInsights.trackPageView(); // Manually call trackPageView to establish the current user/session/pageview
+    ```
+## Set up deployment configuration
+### Deploy Extension solution Tenant wide (All sites share same iKey)
+1. Under your Extension solution config folder, set the skipFeatureDeployment attribute to true in the package-solution.json file. And add "ClientSideInstance.xml" file under "features"/"assets"/"elementManifests".
```

Comment:
Consider continuing the numbers if this is the next step. You may need to have two forks - Option A: 4,5,6... and OPTIONB 4,5,6...

Reply:
Sounds good, I'll update it thanks.